bool initialized_
uint8 data